hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastConfigBeanDefinitionParser.java

```diff
@@ -506,6 +509,35 @@ private void handleReuseAddress(final Node node, final BeanDefinitionBuilder net
             networkConfigBuilder.addPropertyValue("reuseAddress", value);
         }
 
+        private void handleBindAddressProvider(final Node node, final BeanDefinitionBuilder networkConfigBuilder) {
+            final BeanDefinitionBuilder bindAddressProviderConfigBuilder = createBeanBuilder(BindAddressProviderConfig.class);
+            for (Node child : childElements(node)) {
+                if ("properties".equals(cleanNodeName(child))) {
+                    handleProperties(child, bindAddressProviderConfigBuilder);
+                    break;
+                }
+            }
+            final String implementationAttr = "implementation";
+            final String classNameAttr = "class-name";
+            final String implementationValue = getAttribute(node, implementationAttr);
+            final String classNameValue = getAttribute(node, classNameAttr);
+
+            bindAddressProviderConfigBuilder.addPropertyValue("enabled", getBooleanValue(getAttribute(node, "enabled")));
+
+            if (!StringUtil.isNullOrEmpty(implementationValue)) {
```

Comment:
so we favor `implementation` over `class-name`. Does this cause any confusion if both is defined? cc: @googlielmo 

I am not sure we have a standart way of handling this but we might fail fast if both is defined in spring configuration.

Reply:
Right, I'm open to failing fast as well. I'm looking at the other uses and they are similar. E.g. the `handleMapEvictionPolicyConfig` also favours the implementation, otherwise it will choose the class name. The `extractBasicStoreConfig` will add both the implementation and the class name and will not fail fast as well. 
So we have a range of options here. Which one do you think would be best?

Reply:
I am ok with the current implementation. We might add a hint to the documentation for this.

Reply:
Thanks @mmedenjak @mesutcelik I'm okay with the current implementation as well. I agree there should be a standard way to handle when both are defined (e.g., fail fast), but we may look at this after merging this PR.
I opened https://github.com/hazelcast/hazelcast/issues/11566 to keep track of this.